### PR TITLE
Refactor Ride::Open, Ride::Test not to use gGameCommandErrorText

### PR DESCRIPTION
**Pull request by Gymnasiast**
This brings the usage of gGameCommandErrorText down from 62 to 27 occurrences. The rest will follow in other PRs.

### CODE REVIEW

File: src/openrct2/actions/RideSetStatusAction.cpp [hunk 1]
```diff
@@ -93,19 +93,21 @@ GameActions::Result RideSetStatusAction::Query() const
...
+            if (!modeSwitchResult.Successful)
             {
                 res.Error = GameActions::Status::Unknown;
-                res.ErrorMessage = gGameCommandErrorText;
+                res.ErrorMessage = modeSwitchResult.Message;
                 return res;
             }
         }
-        else if (_status == RideStatus::Open)
+        else
```
Comment (Gymnasiast) [inline src/openrct2/actions/RideSetStatusAction.cpp:104]:
```suggestion
        else if (_status == RideStatus::Open)
```

File: src/openrct2/ride/Ride.cpp [hunk 2]
```diff
@@ -2506,7 +2513,7 @@ void ride_set_map_tooltip(TileElement* tileElement)
  *
  *  rct2: 0x006B4CC1
  */
-static StationIndex ride_mode_check_valid_station_numbers(Ride* ride)
+static ResultWithMessage ride_mode_check_valid_station_numbers(Ride* ride)
```
Comment (Gymnasiast) [inline src/openrct2/ride/Ride.cpp:2516]: This always returned a bool, but it was mislabelled as a StationIndex.

**Comment by duncanspumpkin** [comment]
Couldn't you just return a game action result?

**Comment by Gymnasiast** [comment]
I think that would be overkill and rather inefficient. Especially since it would require every helper function to return a massive class where only two of the fields are ever set.

File: src/openrct2/actions/ResultWithMessage.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ *****************************************************************************/
+
+#pragma once
+
+#include "../common.h"
+#include "../localisation/StringIds.h"
+
+struct ResultWithMessage
+{
+    bool Successful;
```
Comment (IntelOrca) [inline src/openrct2/actions/ResultWithMessage.h:17]:
```suggestion
    bool Successful{};
```

**Comment by IntelOrca** [comment]
> I think that would be overkill and rather inefficient. Especially since it would require every helper function to return a massive class where only two of the fields are ever set.

I agree, these are for the helper methods that wrap calls to game actions. Ideally though they would be async or callback. Right now they do not return a result based on the server - only client validation.

**Review by IntelOrca — APPROVED** [review]

**Pull request merged**